crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -131,8 +132,26 @@ impl CumulocityConverter {
         messages_or_err.unwrap_or_else(|error| vec![self.new_error_message(error)])
     }
 
-    pub fn wrap_error(&self, message_or_err: Result<MqttMessage, ConversionError>) -> MqttMessage {
-        message_or_err.unwrap_or_else(|error| self.new_error_message(error))
+    pub fn wrap_errors_with_input(
+        &self,
+        messages_or_err: Result<Vec<MqttMessage>, ConversionError>,
+        input: &MqttMessage,
+    ) -> Vec<MqttMessage> {
+        let error = match messages_or_err {
+            Ok(messages) => return messages,
+            Err(error) => error,
+        };
+
+        let error = MessageConversionError {
+            error,
+            topic: input.topic.name.clone(),
+        };
```

Comment:
How about moving this `MessageConversionError` creation logic into the `convert` function and propagate the same type into `wrap_errors` and `new_error_message` functions as well, since the `new_error_message` function already has the logic below that creates the message on the error topic. That

Reply:
`wrap_errors` is also used in other places where we don't have a message we're converting, for example auto entity registration. Then using the topic of a message that causes auto-registration (for example some measurement) next to any errors about registration could cause confusion

Reply:
Okay, makes sense.

In that case, we can just change the signature of the `new_error_message` function to accept a `std::error::Error` instead of `ConversionError` so that this whole function can be shortened as:

```
message_or_err
    .map_err(
        |error| MessageConversionError {
            error,
            topic: input.topic.name.clone(),
        })
    .unwrap_or_else(|error| self.new_error_message(error))
```